### PR TITLE
[OOT Plugin][Performance] Optimize metadata prepare

### PR DESCRIPTION
**Pull request by ganyi1996ppo**
## Motivation
Optimize metadata preparation for MHA in OOT plugin, for decode only or prefill only case, the metadata prepare process should be compressed to 50~100 us at host side.

image

## Test Plan
gsm8k

## Test Result
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8840|±  |0.0088|
|     |       |strict-match    |     5|exact_match|↑  |0.8749|±  |0.0091|

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

File: atom/plugin/attention.py [hunk 1]
```diff
@@ -329,17 +326,31 @@ def build(
...
         ) = split_ret
+        prefill_only = num_decodes == 0 and num_extends == 0 and num_prefills > 0
+        decode_only = num_decodes > 0 and num_extends == 0 and num_prefills == 0
+        mixed_request = not (prefill_only or decode_only)
 
         query_start_loc_cpu = common_attn_metadata.query_start_loc_cpu
-        seq_lens = common_attn_metadata.seq_lens.cpu()
-        query_lens_cpu = query_start_loc_cpu[1:] - query_start_loc_cpu[:-1]
+        if mixed_request:
+            seq_lens = common_attn_metadata.seq_lens.cpu()
```
Comment (valarLip) [inline atom/plugin/attention.py:311]: vllm don't have something like seq_lens_cpu?

Comment (ganyi1996ppo) [reply atom/plugin/attention.py:311]: vllm deprecate the seq_lens_cpu in `common_metadata` for a while.

File: atom/plugin/attention.py [hunk 2]
```diff
@@ -464,26 +475,32 @@ def build(
...
+                    common_attn_metadata.max_seq_len
+                    if prefill_only
+                    else query_lens_cpu[:num_decodes].max().item()
+                ),
                 query_start_loc=query_start_loc_device - query_start_loc_device[0],
             )
 
-        num_actual_kv_tokens = torch.sum(seq_lens).item()
+        # num_actual_kv_tokens = torch.sum(seq_lens).item()
+        num_actual_kv_tokens = 0
```
[the remark this answers is not in the document]
Comment (ganyi1996ppo) [reply atom/plugin/attention.py:479]: Hardcode to 0 because it's never used in the oot plugin attention backend

**Review by zejunchen-zejun — APPROVED** [review]
LGTM

**Comment by wuhuikx** [comment]
Any comment on this PR? @valarLip  @ChuanLi1101
